### PR TITLE
Add Terraform config for opa gatekeeper constraints

### PR DESCRIPTION
**Pull request by halamix2**
**Description**

Changes proposed in this pull request:

- Add Terraform config for opa gatekeeper constraints
- Add prowjobs for automatically applying the config

**Related issue(s)**

- #7518
/area ci
/kind feature
/retest

### CODE REVIEW

File: opa/terraform/modules/constraints/variables.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+variable "constraint_templates_path" {
+  type        = list(string)
+  description = "Path to the constraint templates directory."
+}
+
+variable "constraints_path" {
+  type        = list(string)
+  description = "Path to the constraints directory."
```
Comment (dekiel) [inline opa/terraform/modules/constraints/variables.tf:9]: These list variables can hold multiple paths.

```suggestion
variable "constraint_templates_path" {
  type        = list(string)
  description = "Paths to the constraint templates directories."
}

variable "constraints_path" {
  type        = list(string)
  description = "Paths to the constraints directories."
```

**Review by dekiel — CHANGES_REQUESTED** [review]
Please remove dummy files.
Please add description to all variables. This is a best practice recommended in Terraform documentation.

File: configs/terraform/environments/prod/output.tf [hunk 2]
```diff
@@ -29,3 +29,19 @@ output "trusted_workload_terraform_executor_k8s_service_account" {
 output "untrusted_workload_terraform_executor_k8s_service_account" {
   value = module.untrusted_workload_terraform_executor_k8s_service_account
 }
+
+
+output "tekton_gatekeeper_constraints" {
+  value     = module.tekton_gatekeeper_constraints
+  sensitive = true
```
[the remark this answers is not in the document]
Comment (halamix2) [reply configs/terraform/environments/prod/output.tf:36]: It now is, after changes to how we handle YAML files

File: opa/terraform/modules/constraints/variables.tf (hunk 1 above)
Comment (halamix2) [reply opa/terraform/modules/constraints/variables.tf]: Done

**Comment by halamix2** [comment]
- TF plan should succeed after #7750 is merged

**Comment by dekiel** [comment]
Wait with merge till terraform-executor prowjob will be fixed in #7755 
/hold

**Review by dekiel — DISMISSED** [review]

**Review by dekiel — APPROVED** [review]

**Pull request merged**